### PR TITLE
sending the appropriate query string parameters in deprovision

### PR DESCRIPTION
**Pull request by arschles**
… and not sending a request body

Fixes https://github.com/kubernetes-incubator/service-catalog/issues/883

### CODE REVIEW

**Comment by arschles** [comment]
cc/ @ljnelson

File: pkg/brokerapi/openservicebroker/open_service_broker_client.go [hunk 1]
```diff
@@ -194,10 +194,17 @@ func (c *openServiceBrokerClient) DeleteServiceInstance(ID string, req *brokerap
...
 	// TODO: Handle the auth
-	resp, err := sendOSBRequest(c, http.MethodDelete, serviceInstanceURL, req)
+	httpReq, err := c.newOSBRequest(http.MethodDelete, serviceInstanceURL, nil)
 	if err != nil {
-		glog.Errorf("Error sending delete service instance request to broker %q at %v: response: %v error: %#v", c.name, serviceInstanceURL, resp, err)
-		return nil, resp.StatusCode, errRequest{message: err.Error()}
+		glog.Errorf("Error creating delete service instance request to broker %q at %v (%#v)", c.name, serviceInstanceURL, err)
+		return nil, 0, errRequest{message: err.Error()}
+	}
+	httpReq.Header.Set("service_id", req.ServiceID)
```
Comment (duglin) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: Why are these HTTP headers? Shouldn't they be query params?

Comment (arschles) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: doh, yes, sorry.

Comment (arschles) [inline pkg/brokerapi/openservicebroker/open_service_broker_client.go]: @duglin fixed in a89810dd1

**Comment by duglin** [comment]
LGTM if CI is happy

**Comment by krancour** [comment]
I LGTM3'ed this and merged it and then discovered a problem with it after the fact. I've opened #887 to revert this PR.

The problem is that the format strings being used to construct URLs were already accounting for these parameters:

```
serviceInstanceDeleteFormatString      = "%s/v2/service_instances/%s?service_id=%s&plan_id=%s"
	serviceInstanceDeleteAsyncFormatString = "%s/v2/service_instances/%s?service_id=%s&plan_id=%s&accepts_incomplete=true"
```

The changes in this PR, imo, honor a more idiomatic means of setting query parameters on an outbound HTTP request, but as it stands now, this code adds query parameters that are already present. That is not ideal.

**Comment by duglin** [comment]
ha that would explain why it appeared to work even when @arschles was setting them as http headers.  good find!

**Pull request merged**